src/cljs/main/broadfcui/net.cljs

```diff
@@ -0,0 +1,43 @@
+(ns broadfcui.net
+  (:require
+  [broadfcui.common.components :as comps]
+  [broadfcui.common.style :as style]
+  [broadfcui.utils :as utils]
+  ))
+
+(defn handle-ajax-response [update-fn]
+  (fn [{:keys [xhr status-code status-text success?]}]
+    (if-let [parsed-response (as-> xhr x
+                                   (aget x "responseText")
+                                   (try (js-invoke js/JSON "parse" x) (catch js/Error e e))
+                                   (when-not (instance? js/Error x)
+                                     (js->clj x :keywordize-keys true)))]
+      (update-fn (utils/restructure success? status-code status-text parsed-response))
+      (update-fn (merge
+                  (utils/restructure success? status-code status-text)
+                  {:parsed-response {:message "Error parsing server response"}
+                   :original-request xhr})))))
+
+(defn render-with-ajax
+  ([ajax-response render-success] (render-with-ajax ajax-response render-success {}))
+  ([{:keys [success? parsed-response] :as ajax-response} render-success
+    {:keys [loading-text rephrase-error error-handler] :as extras}]
+   (assert (not (and rephrase-error error-handler)) "Provide EITHER error-handler OR rephrase-error")
```

Comment:
It's a bit difficult to understand how to use the error handlers. I suggest renaming `error-handler` to `handle-error` (start it with a verb) and passing `ajax-response`. Then, you don't really need `create-error-message-for-code`. See below for notes on usage.